### PR TITLE
DEPR: Enforce deprecation of numeric_only=None in DataFrame aggregations

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
looks like some frame reductions tests are failing

**Comment by rhshadrach** [comment]
> looks like some frame reductions tests are failing

Thanks @jbrockmendel - this got a bit tricky. When the input is numeric but has object dtype, axis=1, and numeric_only=False, in 1.5.x the result is object dtype. When numeric_only=None however, the result is float. I think this behavior originated from #676. It seems odd, but I think a very rare case. I opted to go to with the 1.5.x default (numeric_only=None) behavior here and put a line in the whatsnew. If that looks good, I can open an issue on this for followup.

Also, when working on these fixes, I realized I missed generic.py / series.py for this deprecation. Went unnoticed because `None` is falsey.

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -441,7 +441,7 @@ Removal of prior version deprecations/changes
 - Changed behavior of comparison of a :class:`Timestamp` with a ``datetime.date`` object; these now compare as un-equal and raise on inequality comparisons, matching the ``datetime.datetime`` behavior (:issue:`36131`)
 - Enforced deprecation of silently dropping columns that raised a ``TypeError`` in :class:`Series.transform` and :class:`DataFrame.transform` when used with a list or dictionary (:issue:`43740`)
 - Change behavior of :meth:`DataFrame.apply` with list-like so that any partial failure will raise an error (:issue:`43740`)
--
+- Enforced deprecation of silently dropping columns that raised in DataFrame reductions (:issue:`41480`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.0.0.rst:444]: specific to `numeric_only=None`? or "when `numeric_only` is not specified"?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -10541,25 +10539,22 @@ def _get_data() -> DataFrame:
                 data = self._get_bool_data()
             return data
 
-        numeric_only_bool = com.resolve_numeric_only(numeric_only)
-        if numeric_only is not None or axis == 0:
+        if numeric_only or axis == 0:
```
Comment (jbrockmendel) [inline pandas/core/frame.py:10542]: why can't we go through this path unconditionally?

File: doc/source/user_guide/groupby.rst [hunk 3]
```diff
@@ -1187,12 +1187,6 @@ is only interesting over one column (here ``colname``), it may be filtered
    If you do wish to include decimal or object columns in an aggregation with
    other non-nuisance data types, you must do so explicitly.
 
-.. warning::
```
Comment (jbrockmendel) [inline doc/source/user_guide/groupby.rst:1190]: should this be removed now or wait until a deprecation is enforced in the groupby code?

File: pandas/core/frame.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/frame.py:10542]: Ah, I don't think what I wrote was all too clear, but tried to explain this in https://github.com/pandas-dev/pandas/pull/49551#issuecomment-1304815473. I think we should take this path unconitionality, but there is a behavior change for `numeric_only=False`, `axis=1` and object dtype. I plan to open an issue and do a followup here.

File: doc/source/user_guide/groupby.rst (hunk 3 above)
Comment (rhshadrach) [reply doc/source/user_guide/groupby.rst]: Thanks - got confused on where this was.

File: pandas/core/frame.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/frame.py:10530]: Here is an example to make this more explicit (on main):

```
df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]}, dtype=object)
print(df.sum(axis=1, numeric_only=None))
print(df.sum(axis=1, numeric_only=False))
0    5.0
1    7.0
2    9.0
dtype: float64
0    5
1    7
2    9
dtype: object
```

Taking this path unconditionally would be to always get object dtype. I think that's the right thing to do, but would be a change in the default (numeric_only=None) behavior and plan to handle in a follow up.

Comment (jbrockmendel) [reply pandas/core/frame.py:10530]: makes sense, thanks

**Comment by jbrockmendel** [comment]
needs rebase, otherwise LGTM

File: pandas/core/frame.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/frame.py:10525]: I opened #49603

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Comment by jbrockmendel** [comment]
nice!

**Pull request merged**